crates/sui-sdk/examples/sign_tx_guide.rs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+mod utils;
+use crate::utils::request_tokens_from_faucet;
+use fastcrypto::encoding::Encoding;
+use fastcrypto::hash::HashFunction;
+use fastcrypto::{
+    ed25519::Ed25519KeyPair,
+    encoding::Base64,
+    secp256k1::Secp256k1KeyPair,
+    secp256r1::Secp256r1KeyPair,
+    traits::{EncodeDecodeBase64, KeyPair},
+};
+use rand::{rngs::StdRng, SeedableRng};
+use shared_crypto::intent::{Intent, IntentMessage};
+use sui_sdk::{
+    rpc_types::SuiTransactionBlockResponseOptions,
+    types::{
+        programmable_transaction_builder::ProgrammableTransactionBuilder,
+        transaction::TransactionData,
+    },
+    SuiClientBuilder,
+};
+use sui_types::crypto::Signer;
+use sui_types::crypto::ToFromBytes;
+use sui_types::signature::GenericSignature;
+use sui_types::{
+    base_types::SuiAddress,
+    crypto::{get_key_pair_from_rng, SuiKeyPair},
+};
+
+// This example walks through the Rust SDK usecase in sign-txn.mdx.
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    // set up sui client for the desired network.
+    let sui_client = SuiClientBuilder::default().build_devnet().await?;
+
+    // deterministically generate a keypair, testing only, do not use for mainnet.
```

Comment:
While I think this might be OK for testing purposes, the problem we always have is how to do this for production. Can we please add either a comment or some actual code of what would one do in production? A link to a doc is fine too!

Reply:
for production uses line 49, i can add a comment